### PR TITLE
Add root node to display as parent to all assemblies

### LINKED ISSUE #73: Multiple loaded files have no root in tree
When multiple test assemblies are opened, they are displayed in the tree as separate roots. That is, there is no common root as in V2. Consequently, to run all the tests using the Run button, the user must enable checkboxes and select each root.

Since this version of the gui is intended to be similar to V2, the easy solution would be to provide an artificial root in the tree, which is selected by default. That's my preference.

One alternative would be to provide a separate "Run All" button.

What do folks think?

### PR DESCRIPTION
**Pull request by CharliePoole**
Fixes #73

### CODE REVIEW

File: src/TestCentric/testcentric.gui/Controls/TestSuiteTreeView.cs [hunk 1]
```diff
@@ -599,8 +599,9 @@ public void Load(TestNode topLevelNode)
 
                 try
                 {
-                    foreach (var child in topLevelNode.Children)
-                        AddTreeNodes(Nodes, child, false);
+                    //foreach (var child in topLevelNode.Children)
```
Comment (immeraufdemhund) [inline src/TestCentric/testcentric.gui/Controls/TestSuiteTreeView.cs]: remove commented out code

File: src/TestCentric/testcentric.gui/TestCentricPresenter.cs [hunk 2]
```diff
@@ -391,8 +391,7 @@ public void RunFailedTests()
 
         public void RunTests(TestNode test)
         {
-            EnableRunCommand(false);
-            Model.RunTests(test);
+            RunTests(new TestNode[] { test });
```
Comment (immeraufdemhund) [inline src/TestCentric/testcentric.gui/TestCentricPresenter.cs]: nice :) I like seeing simplified code

File: src/TestModel/model/TestNode.cs [hunk 3]
```diff
@@ -54,13 +54,40 @@ public class TestNode : ITestItem
...
+                    IsSuite = false;
+                    Name = Xml.GetAttribute("name");
+                    FullName = Xml.GetAttribute("fullname") ?? Name;
+                    Type = "TestCase";
+                    TestCount = 1;
+                    RunState = GetRunState();
+                    break;
+            }
+
+            //IsSuite = Xml.Name == "test-suite" || Xml.Name == "test-run";
```
Comment (immeraufdemhund) [inline src/TestModel/model/TestNode.cs]: remove commented out code

File: src/TestModel/model/TestNode.cs [hunk 4]
```diff
@@ -54,13 +54,40 @@ public class TestNode : ITestItem
...
             Xml = xmlNode;
-            IsSuite = Xml.Name == "test-suite" || Xml.Name == "test-run";
             Id = Xml.GetAttribute("id");
-            Name = Xml.GetAttribute("name");
-            FullName = Xml.GetAttribute("fullname");
-            Type = IsSuite ? GetAttribute("type") : "TestCase";
-            TestCount = IsSuite ? GetAttribute("testcasecount", 0) : 1;
-            RunState = GetRunState();
+
+            switch (Xml.Name)
```
Comment (immeraufdemhund) [inline src/TestModel/model/TestNode.cs]: took me a second to realize this is much easier to read. Nice

Comment (immeraufdemhund) [reply src/TestModel/model/TestNode.cs]: i meant the entire switch. made it very clera what was to be shown.

**Comment by CharliePoole** [comment]
I'm also going to fix the item I noted as `HACK` before merge.

File: src/TestCentric/testcentric.gui/Controls/TestSuiteTreeView.cs [hunk 5]
```diff
@@ -1261,6 +1262,10 @@ public class TestNodeCollection : List<TestNode>, ITestItem
 
         public TestFilter GetTestFilter()
         {
+            // HACK
+            if (Count == 1 && this[0].FullName == "TestRun")
+                return TestFilter.Empty;
+
```
Comment (CharliePoole) [inline src/TestCentric/testcentric.gui/Controls/TestSuiteTreeView.cs]: I really don't like this and I just noticed what may be the root cause of the problem. The TestNodeCollection class is defined in the view, kused by the presenter and really belongs in the model!

**Comment by CharliePoole** [comment]
I removed the comments you noted and also restructured. TestNodeCollection is only used in the model, so I moved it there. I took out the HACK and made the test node itself responsible (if it's a test-run node) to return an empty filter when asked for one.

I also fixed up some incorrect comments and removed code for the somewhat related TestSelection class from the solution, but kept in in the repo. Unused code that called it is compiled out using the NEW_GUI define. It would be cleaner to just delete the code, but I want to figure out how to merge the models for the old and new guis before I do that, so NEW_GUI will at least temporarily hide code in the model that is only there for the new gui.

I'd really like to be done with this and work on the new GUI!!!

**Comment by immeraufdemhund** [comment]
sounds nice... I don't see that it got pushed up yet though.

**Comment by CharliePoole** [comment]
Odd. I'll check.

**Pull request merged**